lib/string_reverse.rb

```diff
@@ -1,6 +1,25 @@
 # A method to reverse a string in place.
 # Time complexity: ?
 # Space complexity: ?
+
 def string_reverse(my_string)
-  raise NotImplementedError
+  # raise NotImplementedError
+
+  if my_string == nil
+    return my_string
+  else
+    i = 0
+    while my_string[i] != nil
+      i += 1
+    end #while end
+
+    for j in (0...(i / 2))
+      temp = ""
+
+      temp = my_string[j]
+      my_string[j] = my_string[(i - 1) - j]

```

Comment:
While I understand what you are doing here, `my_string[(i-1)-j]` can be cleaned up to something like `my_string[k]` (used `k` bc you already have an `i` variable). Using a for loop for the first half of the string is what is forcing you to calculate the index of the last char.